docs/spec/concepts.rst

```diff
@@ -3,6 +3,266 @@
+with ``tuple[Any, int]``, and ``tuple[Any, int]`` is consistent with
+``tuple[str, int]``, but ``tuple[int, int]`` is not consistent with
+``tuple[str, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
+is also consistent with ``A``. It is also reflexive: ``A`` is always consistent
+with ``A``.
+
+The assignable-to (or consistent subtyping) relation
+----------------------------------------------------
+
+Given the materialization relation and the subtyping relation, we can define the
+**consistent subtype** relation over all types. A type ``A`` is a consistent
+subtype of a type ``B`` if there exists a materialization ``A'`` of ``A`` and a
+materialization ``B'`` of ``B``, where ``A'`` and ``B'`` are both static types,
+and ``A'`` is a subtype of ``B'``.
+
+Consistent subtyping defines "assignability" for Python.  An expression can be
+assigned to a variable (including passed as a parameter or returned from a
+function) if it is a consistent subtype of the variable's type annotation
+(respectively, parameter's type annotation or return type annotation).
+
+We can say that a type ``A`` is "assignable to" a type ``B`` if ``A`` is a
+consistent subtype of ``B``.
+
+In the remainder of this specification, we will usually prefer the term
+**assignable to** over "consistent subtype of". The two are synonymous, but
+"assignable to" is shorter, and may communicate a clearer intuition to many
+readers.
+
+For example, ``Any`` is assignable to ``int``, because ``int`` is a
+materialization of ``Any``, and ``int`` is a subtype of ``int``. The same
+materialization also gives that ``int`` is assignable to ``Any``.
+
+The assignable-to relation is not generally symmetric, though. If ``B`` is a
+subtype of ``A``, then ``tuple[Any, B]`` is assignable to ``tuple[int, A]``,
+because ``tuple[Any, B]`` can materialize to ``tuple[int, B]``, which is a
+subtype of ``tuple[int, A]``. But ``tuple[int, A]`` is not assignable to
+``tuple[Any, B]``.
+
+Attributes and methods
+----------------------
+
+In Python, we can do more with objects at runtime than just assign them to
+names, pass them to functions, or return them from functions. We can also
+get/set attributes and call methods.
+
+In the Python object model, the operations that can be performed on a value all
+de-sugar to method calls. For example, ``a + b`` is syntactic sugar for either
+``a.__add__(b)`` or ``b.__radd__(a)``.
+
+For a static type checker, accessing ``a.foo`` is a type error unless all
+possible objects in the set represented by the type of ``a`` have the ``foo``
+attribute.
+
+If all objects in the set represented by the fully static type ``A`` have a
+``foo`` attribute, we can say that the type ``A`` has the ``foo`` attribute.
+
+If the type ``A`` of ``a`` in ``a.foo`` is a gradual type, it may not represent
+a single set of objects. In this case, ``a.foo`` is not a type error if and
+only if there exists a materialization of ``A`` which has the ``foo``
```

Comment:
Or has a `__[get|set|del]attr__` method. Once again, not sure this is worth adding.